File: ElectionVote/ElectionVote/src/main/java/Backend/ElectionVote/service/PollingPlaceService.java
package Backend.ElectionVote.service;

import Backend.ElectionVote.dto.PollingPlaceCreateRequest;
import Backend.ElectionVote.dto.PollingPlaceDto;
import Backend.ElectionVote.dto.PollingPlaceUpdateRequest;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.UUID;

public interface PollingPlaceService {

    PollingPlaceDto create(PollingPlaceCreateRequest req);

    PollingPlaceDto update(UUID id, PollingPlaceUpdateRequest req);

    PollingPlaceDto get(UUID id);

    List<PollingPlaceDto> listByCenter(UUID centerId);

    PollingPlaceDto setActive(UUID id, boolean active);

    void delete(UUID id);

    Page<PollingPlaceDto> list(int page, int size, String q,
                               UUID countyId, UUID districtId, UUID centerId,
                               Boolean active);


}

File: frontend/Old Folders/shared/services/pollingPlaceService.ts
// // src/shared/services/pollingPlaceService.ts
// // ------------------------------------------------------
// // Service for Polling Places.
// // Backend endpoint:
// //   GET /api/polling-places/by-center/{centerId}
// // where centerId is a UUID.
// // ------------------------------------------------------

// import { apiClient } from "../lib/apiClient";

// export interface PollingPlaceDto {
//   placeId: string;

//   centerId: string;
//   centerCode: string;
//   centerName: string;

//   districtId: string;
//   districtName: string;

//   countyId: string;
//   countyName: string;

//   placeNumber: number;
//   code: string;
//   label?: string | null;

//   active: boolean;
// }

// /** Load polling places for a given centerId (UUID). */
// export async function fetchPollingPlacesByCenter(
//   centerId: string
// ): Promise<PollingPlaceDto[]> {
//   const res = await apiClient.get<PollingPlaceDto[]>(
//     `/polling-places/by-center/${centerId}`
//   );
//   return res.data ?? [];
// }
